### PR TITLE
[infra] Prevent mounting local checkout to $SRC + clean up Dockerfiles (#1699).

### PR DESCRIPTION
**Pull request by Dor1s**

### CODE REVIEW

**Comment by Dor1s** [comment]
Seems to work locally, so I'm landing for now. Oliver, please take a look whether you're fine with this change.

File: infra/helper.py [hunk 1]
```diff
@@ -295,9 +295,10 @@ def _env_to_docker_args(env_list):
   return sum([['-e', v] for v in env_list], [])
 
 
-def _workdir_from_dockerfile(dockerfile_path):
-  """Parse WORKDIR from the Dockerfile."""
-  WORKDIR_REGEX = re.compile(r'\s*WORKDIR\s*([^\s]+)')
+def _workdir_from_dockerfile(project_name):
+  """Parse WORKDIR from the Dockerfile for the given project."""
+  WORKDIR_REGEX = re.compile(r'\s*WORKDIR\s*(\$SRC/[^\s]+)')
```
Comment (oliverchang) [inline infra/helper.py]: why is the addition of $SRC needed in the regex? There are lots of Dockerfiles without $SRC in their WORKDIR command.

Comment (oliverchang) [reply infra/helper.py]: For example this is wrong:

```
>>> import helper
>>> helper._workdir_from_dockerfile('firefox')
'/src/firefox'
```
```
Should be:
'/src/mozilla-central'
```

Comment (Dor1s) [reply infra/helper.py]: Yeah :( let me upload another fix.

**Pull request merged**